File: JVA/ACM/Tree.java
package JVA.ACM;

import java.util.Queue;
import java.util.LinkedList;

public class Tree {
    int val;
    TreeNode root;

    // Constructor - use bfs to create a tree
    public Tree(Integer[] arr) {
        if (arr == null || arr.length == 0) {
            throw new IllegalArgumentException("Array cannot be empty");
        }
        root = new TreeNode(arr[0]);
        int i = 1;
        int n = arr.length;
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);
        while (i < n) {
            TreeNode cur = queue.poll();
            if (arr[i] != null) {
                cur.left = new TreeNode(arr[i]);
                queue.offer(cur.left);
            }
            i++;
            if (i < n && arr[i] != null) {
                cur.right = new TreeNode(arr[i]);
                queue.offer(cur.right);
            }
            i++;
        }
    }
}

File: JS/ACM/tree.js

class Node {
    constructor(val) {
        this.val = val;
        this.left = this.right = null;
    }
}

export default class Tree {
    constructor(arr) {
        this.root = new Node(arr[0]);
        let queue = [this.root];
        for (let i = 1; i < arr.length; i++) {
            const cur = queue.shift();
            if (arr[i] !== null) {
                cur.left = new Node(arr[i]);
                queue.push(cur.left);
            }
            if (arr[++i] !== null) {
                cur.right = new Node(arr[i]);
                queue.push(cur.right);
            }
        }
    }
    preOrderToString(node = this.root) {
        if (!node) return "";
        return node.val + " " + this.preOrderToString(node.left) + this.preOrderToString(node.right);
    }
    inOrderToString(node = this.root) {
        if (!node) return "";
        return this.inOrderToString(node.left) + node.val + " " + this.inOrderToString(node.right);
    }
    postOrderToString(node = this.root) {
        if (!node) return "";
        return this.postOrderToString(node.left) + this.postOrderToString(node.right) + node.val + " ";
    }
}
